### PR TITLE
optimization hint inner 1658

### PR DESCRIPTION
**Pull request by ylinzhu**
Reason:  
  Improve 
Type:  
  Improve  
Influences：  
   fix xx

### CODE REVIEW

File: src/main/java/com/actiontech/dble/plan/optimizer/HintPlanInfo.java [hunk 1]
```diff
@@ -5,34 +5,32 @@
...
     private boolean left2inner = false;
     private boolean right2inner = false;
     private boolean in2join = false;
 
-    public HintPlanInfo(@Nonnull List<HintPlanNodeGroup> groups) {
-        this.groups = groups;
-    }
+    private HashMap<String, Set<HintPlanNode>> dependMap = Maps.newHashMap();
+    private HashMap<String, Set<HintPlanNode>> erMap = Maps.newHashMap();
+    private HashMap<String, HintPlanParse.Type> hintPlanNodeMap = Maps.newHashMap();
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/HintPlanInfo.java:25]: must use linkedmap

**Review by dcy10000 — APPROVED** [review]

File: src/main/java/com/actiontech/dble/plan/optimizer/HintPlanInfo.java (hunk 1 above)
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/plan/optimizer/HintPlanInfo.java:25]: must use linkedmap

**Review by dcy10000 — CHANGES_REQUESTED** [review]

**Comment by dcy10000** [comment]
d|(a&b)&c can't parsed

**Review by PanternBao — APPROVED** [review]
lgtm

**Review by dcy10000 — APPROVED** [review]
lgtm

**Pull request merged**